packages/puppeteer/test/axePuppeteer.test.ts

```diff
@@ -0,0 +1,711 @@
+import * as path from 'path';
+import { assert, expect } from 'chai';
+import Puppeteer, { Browser, Page } from 'puppeteer';
+import { createServer, Server } from 'http';
+import * as sinon from 'sinon';
+import testListen from 'test-listen';
+import AxePuppeteer from '../src/index';
+import {
+  startServer,
+  puppeteerArgs,
+  expectAsync,
+  expectAsyncToNotThrow
+} from './utils';
+
+type SinonSpy = sinon.SinonSpy;
+
+declare global {
+  interface Window {
+    parallel?: boolean;
+  }
+}
+
+describe('AxePuppeteer', function () {
+  let browser: Browser;
+  let page: Page;
+  let server: Server;
+  let addr: string;
+  this.timeout(10000);
+
+  let axeSource: string;
+  let axeCrasherSource: string;
+
+  before(async () => {
+    const axePath = require.resolve('axe-core');
+    axeSource = fs.readFileSync(axePath, 'utf8');
+    const axeCrashPath = path.resolve(
+      __dirname,
+      './fixtures/external/axe-crasher.js'
+    );
+    axeCrasherSource = fs.readFileSync(axeCrashPath, 'utf8');
+  });
+
+  before(async () => {
+    const args = puppeteerArgs();
+    browser = await Puppeteer.launch({ args });
+    ({ server, addr } = await startServer());
+  });
+
+  after(async () => {
+    server.close();
+    await browser.close();
+  });
+
+  beforeEach(async () => {
+    page = await browser.newPage();
+  });
+
+  afterEach(async () => {
+    await page.close();
+  });
+
+  it('runs in parallel', async () => {
+    // Just to prove Puppeteer runs scripts in parallel,
+    // and so axe-core/puppeteer should too
+    await page.goto(`${addr}/external/index.html`);
+    const p1 = page.evaluate(() => {
+      window.parallel = true;
+      return new Promise(res => {
+        setTimeout(() => {
+          window.parallel = false;
+          res('parallel');
+        }, 1000);
+      });
+    });
+    const p2 = page.evaluate(() => window.parallel);
+    const out = await Promise.all([p1, p2]);
+    expect(out).to.deep.equal(['parallel', true]);
+  });
+
+  describe('constructor', () => {
+    it('accepts a Page', async () => {
+      await page.goto(`${addr}/external/index.html`);
+      const axePup = new AxePuppeteer(page);
+      await expectAsyncToNotThrow(() => axePup.analyze());
+    });
+
+    it('accepts a Frame', async () => {
+      await page.goto(`${addr}/external/index.html`);
+      const axePup = new AxePuppeteer(page.mainFrame());
+      await expectAsyncToNotThrow(() => axePup.analyze());
+    });
+
+    it('accepts custom axe-core source', async () => {
+      const axeSource = `
+        window.axe = {
+          run: () => new Promise(resolve => resolve({})),
+          configure: () => {}
+        }
+      `;
+      await page.goto(`${addr}/external/index.html`);
+      const evalSpy: SinonSpy = sinon.spy(page.mainFrame(), 'evaluate');
+      await new AxePuppeteer(page, axeSource).analyze();
+      assert(evalSpy.calledWith(axeSource));
+    });
+    // TODO: Defaults to using the bundled axe-core source
```

Comment:
Do we need to worry about this TODO in this pr?

Reply:
Was copied from the other source... probably shouldn't be here. I'll just take it out.